src/unstract/sdk/index.py

```diff
@@ -186,123 +187,138 @@ def index_file(
+            for item in full_text:
+                text = item["text_contents"]
+                self.tool.stream_log("Indexing file...")
+                document = Document(
+                    text=text,
+                    doc_id=doc_id,
+                    metadata={"section": item["section"]},
                 )
-                nodes = parser.get_nodes_from_documents(documents, show_progress=True)
-                node = nodes[0]
-                node.embedding = embedding.get_query_embedding(" ")
-                vector_db.add(doc_id, nodes=[node])
-                self.tool.stream_log("Added node to vector db")
-            else:
-                storage_context = vector_db.get_storage_context()
-                parser = SimpleNodeParser.from_defaults(
-                    chunk_size=chunk_size, chunk_overlap=chunk_overlap
-                )
-
-                self.tool.stream_log("Adding nodes to vector db...")
+                document.id_ = doc_id
+                documents.append(document)
+            self.tool.stream_log(f"Number of documents: {len(documents)}")
 
-                vector_db.get_vector_store_index_from_storage_context(
-                    documents,
-                    storage_context=storage_context,
-                    show_progress=True,
-                    embed_model=embedding,
-                    node_parser=parser,
+            try:
+                if chunk_size == 0:
+                    parser = SimpleNodeParser.from_defaults(
+                        chunk_size=len(documents[0].text) + 10, chunk_overlap=0
+                    )
+                    nodes = parser.get_nodes_from_documents(
+                        documents, show_progress=True
+                    )
+                    node = nodes[0]
+                    node.embedding = embedding.get_query_embedding(" ")
+                    vector_db.add(doc_id, nodes=[node])
+                    self.tool.stream_log("Added node to vector db")
+                else:
+                    storage_context = vector_db.get_storage_context()
+                    parser = SimpleNodeParser.from_defaults(
+                        chunk_size=chunk_size, chunk_overlap=chunk_overlap
+                    )
+                    self.tool.stream_log("Adding nodes to vector db...")
+                    # TODO: Phase 2:
+                    # Post insertion to VDB, use query using doc_id and
+                    # store all the VDB ids to a table against the doc_id
+                    # During deletion for cases where metadata filtering
+                    # does not work, these ids can be used for direct deletion
+                    # This new table will also act like an audit trail for
+                    # all nodes that were added to the VDB by Unstract
+                    # Once this is in place, the overridden implementation
+                    # of prefixing ids with doc_id before adding to VDB
+                    # can be removed
```

Comment:
@gaya3-zipstack Is the reason why we need to do this because the vector store query may fail to return relevant results?

Reply:
This is the proposed design  delete vector db records while indexing/re-indexing so that we can handle cases like Pinecone where metadata filtering does not work.

Reply:
https://zipstack.atlassian.net/browse/UN-1352

Reply:
On a general note, we might need to refrain from doing this and trying to move away from it in the long run, as it would be better not to fiddle with LlamaIndex internal bookkeeping at all, and instead focus on various indexing and retrieval strategies offered by the framework.

Ideally we should also aggressively flag any LlamaIndex issues found for a particular provider, and maybe even raise patches of our own.